modules/virt-accessing-vmi-yaml-ssh.adoc

```diff
@@ -0,0 +1,140 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vmi-yaml-ssh_{context}"]
+= Accessing a virtual machine instance via SSH with YAML configurations
+
+You can enable a SSH connection to a virtual machine instance (VMI) without the need to run the `virtctl expose` command. When the YAML file for the VMI and the YAML file for the service are configured and applied,  the service forwards the SSH traffic to the VMI.
+
+The following examples show the configurations for the VMI's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* You must be in the same project as the VMI.
+
+.Procedure
+. In the YAML file for the VMI, add the label and a value for exposing the service for SSH connections. Add the port number and enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+kind: VirtualMachineInstance
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-ssh <1>
+  name: vmi-ssh
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {} <2>
+        name: testmasquerade <3>
+        ports:
```

Comment:
Why this is needed?
We only create a service without specify any port under masquerade interface and ssh works.

```
spec:
  clusterIP: 172.30.158.232
  clusterIPs:
  - 172.30.158.232
  externalTrafficPolicy: Cluster
  ports:
  - nodePort: 32040
    port: 22
    protocol: TCP
    targetPort: 22
  selector:
    kubevirt.io/domain: vma-1618386812-2222579
  sessionAffinity: None
  type: NodePort

```

Reply:
+1 specifying the ports here is redundant and has different effects that `virtctl expose` we want to mimic.

Reply:
@phoracek , @myakove - I am deferring your question to @alonSadan - he provided guidance to use the service file. @alonSadan - could you please respond to the question asked here about creating a service without specifying any port under masquerade interface for ssh?

Reply:
@phoracek or @myakove - Noting the following from your input.

--Remove the following portion from the YAML file example:
ports:
        - name: ssh
          port: 22 <4>
          protocol: TC

Reply:
The ports section is redounded indeed. My bad.